pandas/core/arrays/masked.py

```diff
@@ -507,7 +507,7 @@ def to_numpy(
         else:
             with warnings.catch_warnings():
                 warnings.filterwarnings("ignore", category=RuntimeWarning)
-                data = self._data.astype(dtype, copy=copy)
+                data = np.array(self._data, dtype=dtype, copy=copy)
```

Comment:
Made this change together with allowing to get here with `copy=False`, to ensure if `copy=False` is not actually possible depending on the passed `dtype`, it still properly errors

Reply:
This change (together with the `self._hasna` change below) make sense to me.  But unfortunately, `to_numpy(copy=False)` change the meaning of `copy=False` this way.  And I think this is public API?

Maybe better to special case `_hasna` in the branch below?  (could even just ignore `dtype` with a comment that NumPy is fine with that).

Reply:
Yeah, will have to update this to not change `to_numpy(copy=False)`, as that is indeed public API we want to keep working that way.